client/verta/tests/test_versioning/test_dataset.py

```diff
@@ -358,6 +359,34 @@ def test_not_to_s3_dir(self, commit, in_tempdir):
         dirpath = dataset.download("s3://")
         assert "s3:" not in pathlib2.Path(dirpath).parts
 
+    def test_download_all(self, commit):
+        s3 = pytest.importorskip("boto3").client('s3')
+
+        bucket = "verta-versioned-bucket"
+        dirname = "tiny-files/"
+        s3_folder = "s3://{}/{}".format(bucket, dirname)
+
+        # get files' contents directly from S3 for reference
+        reference_dir = "reference/"
+        for s3_obj in s3.list_objects_v2(Bucket=bucket, Prefix=dirname)['Contents']:
+            key = s3_obj['Key']
+            filepath = os.path.join(reference_dir, bucket, key)
+            pathlib2.Path(filepath).parent.mkdir(parents=True, exist_ok=True)  # create parent dirs
+
+            s3.download_file(bucket, key, filepath)
+
+        # commit dataset blob
+        blob_path = "data"
+        dataset = verta.dataset.S3(s3_folder, enable_mdb_versioning=True)
+        commit.update(blob_path, dataset)
+        commit.save("Version data.")
+        dataset = commit.get(blob_path)
+
+        dirpath = dataset.download()
+        assert dirpath == os.path.abspath(_dataset.DEFAULT_DOWNLOAD_DIR)
+
+        assert_dirs_match(dirpath, reference_dir)
```

Comment:
This function is added in [#866](https://github.com/VertaAI/modeldb/pull/866/files#diff-2b17498431c2e2c225a23e8b2a7a7450R18)